### PR TITLE
Storage efficient docker build image, CI, linter improvements

### PR DESCRIPTION
**Pull request by noslav**
* Updates docker workflow to reduce image size by 95% (2GB to 90MB)
* Lints code and fixes errors
* Updates CI to use `hadolint`, `golangci-lint`
* Updates ambiguous name "server" to ipfs-server

| REPOSITORY |                            TAG   |    IMAGE ID   |    CREATED     |    SIZE |
| ------------- | --------------------- | ------------ | -------------- | ------- |
| gcr.io/covalent-project/ipfs-pinner  | local |     feb4387598d1  | 2 hours ago   |  90.1MB |
| gcr.io/covalent-project/ipfs-pinner  | latest |    c46ba5e79b2c |  11 hours ago |   2.02GB |

### CODE REVIEW

**Comment by kitti-katy** [comment]
I have Mac M1 and it throws 
`qemu: uncaught target signal 11 (Segmentation fault) - core dumped`

when I run 
`docker run -p 3000:3000 -i --name pinner-instance gcr.io/covalent-project/ipfs-pinner:latest;`

**Comment by kitti-katy** [comment]
also I see 
`gcr.io/covalent-project/ipfs-pinner:latest`
is only 49.7 MB not  2.02GB

**Comment by noslav** [comment]
> I have Mac M1 and it throws `qemu: uncaught target signal 11 (Segmentation fault) - core dumped`
> 
> when I run `docker run -p 3000:3000 -i --name pinner-instance gcr.io/covalent-project/ipfs-pinner:latest;`

delete all related images from your local first? and then try fresh image `docker pull gcr.io/covalent-project/ipfs-pinner:latest@sha256:d4b347bb48a96029df818abc6697b687d1ecf7dcca8eba32fd1048367209c49f`

**Comment by noslav** [comment]
@kitti-katy  @sudeepdino008  add an`.env` file to your docker run `docker run --env-file .env <image_id>` in `.env` add some random web3_jwt or an actual one `WEB3_JWT=ABC....XYZ` in either case you should see something like

```
generating 2048-bit RSA keypair...done
peer identity: QmPWG9HmG6GwgT5mGZA4AqhCCRzECHewjMVxsgNJ9ssjmv
2022/11/01 20:14:27 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
2022/11/01 20:14:27 Listening...
```

**Comment by sudeepdino008** [comment]
@noslav @kitti-katy I've added instructions for `docker run` in the README.md file. It contains the required port exports and docker volumes setup.

**Comment by sudeepdino008** [comment]
I'm not able to test because I cannot pull (there's some authorization error...checking with @cloggo)

```bash

>> ipfs-pinner git:(main) ✗
➜ docker pull gcr.io/covalent-project/ipfs-pinner@sha256:d4b347bb48a96029df818abc6697b687d1ecf7dcca8eba32fd1048367209c49f
Error response from daemon: unauthorized: You don't have the needed permissions to perform this operation, and you may have invalid credentials. To authenticate your request, follow the steps in: https://cloud.google.com/container-registry/docs/advanced-authentication
```

Logged in using `gcloud auth login`, still the same error.

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
+WORKDIR /build
+COPY . .
+RUN go mod download && CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -ldflags="-s -w" -o ipfs-server ./server/main.go
+
+# Runtime -  second phase.
+FROM alpine:3.15.0
+RUN mkdir /app
+WORKDIR /app
+RUN apk update && apk add --no-cache bash=5.1.16-r0
+COPY --from=builder /build/ipfs-server /app
```
Comment (sudeepdino008) [inline Dockerfile:13]: awesome!

**Review by sudeepdino008 — APPROVED** [review]

**Pull request merged**